pubsub/rabbitmq/rabbitmq.go

```diff
@@ -100,6 +100,27 @@ type rabbitMQConnectionBroker interface {
 	Close() error
 }
 
+type ErrorCollection struct {
+	errors []error
+	mux    sync.Mutex
+}
+
+func (c *ErrorCollection) Append(e error, stopCh chan struct{}) {
+	c.mux.Lock()
+	if len(c.errors) == 0 {
+		stopCh <- struct{}{}
```

Comment:
Reading from a closed channel is safe and unblocked, so `stopCh <- struct{}{}` can be safely deleted.